testbed/tests/app/test_desktop.py

```diff
@@ -370,25 +368,45 @@ async def test_current_window(app, app_probe, main_window):
     window2.content = toga.Box(style=Pack(background_color=CORNFLOWERBLUE))
     window3.content = toga.Box(style=Pack(background_color=FIREBRICK))
 
-    # We don't need to probe anything window specific; we just need
-    # a window probe to enforce appropriate delays.
-    window1_probe = window_probe(app, window1)
-
     window1.show()
     window2.show()
     window3.show()
 
-    await window1_probe.wait_for_window("Extra windows added")
+    await main_window_probe.wait_for_window("Extra windows added")
+
+    # When a window without any dialog is made the current_window,
+    # then `app.current_window` should return the specified window.
+    app.current_window = window1
+    await main_window_probe.wait_for_window("Window 1 is current")
+    if app_probe.supports_current_window_assignment:
+        assert app.current_window == window1
+
+    # When a dialog is in focus, `app.current_window` should
+    # return the window from which the dialog was initiated.
+
+    window2_info_dialog = toga.InfoDialog("Second Window Info", "Some info")
+    app_probe.setup_info_dialog_result(window2_info_dialog)
 
     app.current_window = window2
-    await window1_probe.wait_for_window("Window 2 is current")
+    window2_dialog_task = app.loop.create_task(window2.dialog(window2_info_dialog))
+    await main_window_probe.wait_for_window("Window 2 is current")
     if app_probe.supports_current_window_assignment:
         assert app.current_window == window2
+    await app_probe.redraw("select 'OK'")
+    # Cancel the task to avoid dangling
+    window2_dialog_task.cancel()
```

Comment:
I'm unclear why this cancel is needed. What sort of "dangling" can occur? Why is this style of cancel needed here when it isn't needed on any other dialog test?

Reply:
I was getting error for the app still having dangling tasks when exiting the test. On deeper inspection, the error was probably telling me that the dialog is not dismissed, even after calling `await app_probe.redraw("select 'OK'")`